openlibrary/plugins/importapi/code.py

```diff
@@ -211,9 +211,11 @@ def POST(self):
             actual_length = int(rec.leader()[:MARC_LENGTH_POS])
             edition['source_records'] = 'marc:%s/%s:%s:%d' % (ocaid, filename, offset, actual_length)
 
-            #TODO: Look up URN prefixes to support more sources
-            prefix = 'trent'
-            edition['local_id'] = ['urn:%s:%s' % (prefix, _id) for _id in rec.get_fields('001')]
+            #TODO: Look up URN prefixes to support more sources, extend openlibrary/catalog/marc/sources?
+            if ocaid == 'OpenLibraries-Trent-MARCs':
```

Comment:
Longer term, it would be nice to have a dictionary of library local IDs (mapped to their handling funcitons) and or self-documenting classes to group logic.

Looks good to me for now

Reply:
openlibrary/catalog/marc/sources.py
openlibrary/catalog/marc/sources
openlibrary/templates/history/sources.html
Seem to be the current locations where this info is stored.

openlibrary/catalog/marc/sources is a really simple 2 col tsv, so will probably need a reformat.
I'm not completely sure what else relies of the sources. Needs further investigation